### ISSUE TITLE
running with jest, instead of mocha

### ISSUE THREAD

**Issue by chr33s** [tsedio/tsed-getting-started#1]
Hi I forked the repo to [github.com/chr33s/tsed-getting-started](/chr33s/tsed-getting-started) and checked out the ```v5.21.0``` branch to ```chr33s/v5.21.0```  to run the tests with jest instead of mocha.

But running the fork with ```npm test``` the routes aren't mounted, as shown by the console.log
```
no routes mounted ---> []
```

Any ideas?
Thanks

**Comment by chr33s** [comment]
I updated the fork to exclude ts-jest based on https://github.com/TypedProject/ts-express-decorators/issues/477 , with same issue.

**Comment by Romakita** [comment]
Hello @chr33s,
Unit works fine with mocha :)
I'll take a look on your fork to see if you've missed something 
Romain

**Comment by Romakita** [comment]
It's a bad idea to use jest as unit test framework to test backend code. Jest is designed to test front-end code. You should use ts-jest if you really want Jest as unit test framework :)

**Comment by chr33s** [comment]
@Romakita thanks, still new to the codebase and decorators in general.

**Comment by Romakita** [comment]
Ok I found the problem. You use babel to transpile code, and babel change the tree directories. To solve it you have to import manually the controllers on server. I'll send a pull request on your fork :)

**Comment by Romakita** [comment]
I can't make a PR so here the fix:

```typescript
import "reflect-metadata";
import {GlobalAcceptMimesMiddleware, ServerLoader, ServerSettings} from "@tsed/common";
import "@tsed/swagger";
import {$log} from "ts-log-debug";
import {RestCtrl} from "./controllers/RestCtrl";
import {CalendarsCtrl} from "./controllers/calendars/CalendarsCtrl";

const cookieParser = require("cookie-parser");
const bodyParser = require("body-parser");
const compress = require("compression");
const methodOverride = require("method-override");
const rootDir = __dirname;

@ServerSettings({
  rootDir,
  acceptMimes: ["application/json"],
  mount: {
    "/rest": [RestCtrl, CalendarsCtrl] // Add controllers manually
  },
  logger: {
    debug: true,
    logRequest: true,
    requestFields: ["reqId", "method", "url", "headers", "query", "params", "duration"]
  },
  swagger: {
    path: "/api-docs"
  },
  calendar: {
    token: true
  },
  debug: true
})
export class Server extends ServerLoader {
  /**
   * This method let you configure the middleware required by your application to works.
   * @returns {Server}
   */
  $onMountingMiddlewares(): void | Promise<any> {
    this
      .use(GlobalAcceptMimesMiddleware)
      .use(cookieParser())
      .use(compress({}))
      .use(methodOverride())
      .use(bodyParser.json())
      .use(bodyParser.urlencoded({
        extended: true
      }));

    return null;
  }

  $onReady() {
    $log.debug("Server initialized");
  }

  $onServerInitError(error): any {
    $log.error("Server encounter an error =>", error);
  }
}
```

**Comment by Romakita** [comment]
And fixed test:

```typescript
import {$log, ExpressApplication, RouteService} from "@tsed/common";
import {TestContext} from "@tsed/testing";
import SuperTest from "supertest";
import {Server} from "../../../../src/Server";

describe("Calendars", () => {
  let request;
  // bootstrap your expressApplication in first
  beforeAll(TestContext.bootstrap(Server));
  beforeAll(TestContext.inject([ExpressApplication], (expressApplication: ExpressApplication) => {
    request = SuperTest(expressApplication);
  }));
  beforeAll(TestContext.inject([RouteService], (routeService: RouteService) => {
    console.log("no routes mounted --->", routeService.getAll());
  }));
  afterAll(TestContext.reset);

  // then run your test
  describe("GET /rest/calendars", () => {
    it("should return all calendars", async () => {
      const response = await request.get("/rest/calendars").expect(200);
      expect(response.body).toBeInstanceOf(Array);
    });
  });
});
```

**Comment by chr33s** [comment]
@Romakita thanks, your time and reply is greatly, appreciated.

**Comment by chr33s** [comment]
related to https://github.com/TypedProject/ts-express-decorators/issues/477

**Comment by tomer-amir-vonage** [comment]
@chr33s @Romakita This only fixes the issue with `mount`, but when using `componentScan` it still doesn't work since all services and middlewares aren't loaded properly...
I am currently trying to find out if there is another way to fix this, maybe from the `ts-jest` side of things, but I am not sure...

Any help will be welcome :)

**Comment by tomer-amir-vonage** [comment]
@chr33s the issue was fixed in `5.28.0` by [#634](https://github.com/TypedProject/ts-express-decorators/pull/634)
